### PR TITLE
feat: add shuffle parameter to train_test_split

### LINKED ISSUE #673: Add `shuffle: bool` parameter in `train_test_split`
## 🚀 Feature

`train_test_split` function randomly shuffles the dataset and then splits it.

- Add a parameter `shuffle: bool`, which can be used to split without shuffling.

### Motivation

check this [thread](https://discord.com/channels/1077906959069626439/1402295130391576747)

### PR DESCRIPTION
**Pull request by otogamer**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lit-data/blob/main/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes #673.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/utilities/test_train_test_split.py [hunk 1]
```diff
@@ -108,3 +110,28 @@ def test_train_test_split_with_streaming_dataloader(tmpdir, compression):
...
+
+    my_streaming_dataset = StreamingDataset(input_dir=str(tmpdir))
+
+    with patch("litdata.utilities.train_test_split.shuffle_lists_together") as mock_shuffle:
+        train_test_split(my_streaming_dataset, splits=[0.75, 0.25])
+        assert mock_shuffle.call_count == 1, "shuffle=True should call shuffle_lists_together"
+
+    with patch("litdata.utilities.train_test_split.shuffle_lists_together") as mock_shuffle:
+        train_test_split(my_streaming_dataset, splits=[0.75, 0.25], shuffle=False, seed=42)
+        assert mock_shuffle.call_count == 0, "shuffle=False should not call shuffle_lists_together"
```
Comment (deependujha) [inline tests/utilities/test_train_test_split.py]: a better approach imo will be test the desired behaviour and not just checking `call_count`.

Comment (otogamer) [reply tests/utilities/test_train_test_split.py]: I certainly think so.
I made it so that you can check whether it is shuffled or not by comparing the return values.

c53561b
I hope this is a good fix.

**Review by bhimrazy — APPROVED** [review]
LGTM. 🚀
Appreciate your work, @otogamer — looking forward to more amazing contributions!

**Review by Borda — APPROVED** [review]

**Comment by deependujha** [comment]
🎉 Thanks a lot for your contribution, @otogamer!  Looking forward to more awesome contributions from you in the future! ⚡️

**Comment by bhimrazy** [comment]
Hi @otogamer!,
If you're not already there, pls feel free to join our [LightningAI Discord community](https://discord.gg/VptPCZkGNa) — we'd love to have you around!

You're welcome to start any discussions or ask questions in the `#litdata` channel. Looking forward to collaborating more with you! ⚡
